lib/auth.coffee

```diff
@@ -64,20 +59,19 @@ exports.whoami = (callback) ->
 # @param {module:resin.auth~authenticateCallback} callback - callback
 #
 # @example
-#	resin.auth.authenticate credentials, (error, token, username) ->
+#	resin.auth.authenticate credentials, (error, token) ->
 #		throw error if error?
-#		console.log("My username is: #{username}")
 #		console.log("My token is: #{token}")
 ###
 exports.authenticate = (credentials, callback) ->
-	request.request
-		method: 'POST'
-		url: '/login_'
-		json: credentials
-	, (error, response) ->
-		return callback(error) if error?
-		savedToken = response?.body
-		return callback(null, savedToken, credentials.username)
+	Promise.fromNode (callback) ->
+		request.request
```

Comment:
Sadly  `.spread' did not work on this function:

```sh
TypeError: expecting an array, a promise or a thenable. See http://goo.gl/s8MMhc
```

So I had to use `.then` here, but it worked in `.register()`.